src/Platforms/CUDA/cuBLAS_inhouse.cu

```diff
@@ -0,0 +1,293 @@
+                                   const char trans,
+                                   const int m,
+                                   const int n,
+                                   const double* alpha,
+                                   const double* const A[],
+                                   const int lda,
+                                   const double* const x[],
+                                   const int incx,
+                                   const double* beta,
+                                   double* const y[],
+                                   const int incy,
+                                   const int batch_count)
+{
+  return gemv_batched_impl(handle, trans, m, n, alpha, A, lda, x, incx, beta, y, incy, batch_count);
+}
+
+cuBLAS_inhouse_status gemv_batched(cuBLAS_inhouse_handle& handle,
+                                   const char trans,
+                                   const int m,
+                                   const int n,
+                                   const std::complex<float>* alpha,
+                                   const std::complex<float>* const A[],
+                                   const int lda,
+                                   const std::complex<float>* const x[],
+                                   const int incx,
+                                   const std::complex<float>* beta,
+                                   std::complex<float>* const y[],
+                                   const int incy,
+                                   const int batch_count)
+{
+  //return gemv_batched_impl(handle, trans, m, n, (const cuComplex*)alpha, (const cuComplex* const*)A, lda, (const cuComplex* const*)x, incx, (const cuComplex*)beta, (cuComplex* const*)y, incy, batch_count);
+  return cudaSuccess;
+}
+
+cuBLAS_inhouse_status gemv_batched(cuBLAS_inhouse_handle& handle,
+                                   const char trans,
+                                   const int m,
+                                   const int n,
+                                   const std::complex<double>* alpha,
+                                   const std::complex<double>* const A[],
+                                   const int lda,
+                                   const std::complex<double>* const x[],
+                                   const int incx,
+                                   const std::complex<double>* beta,
+                                   std::complex<double>* const y[],
+                                   const int incy,
+                                   const int batch_count)
+{
+  //return gemv_batched_impl(handle, trans, m, n, (const cuDoubleComplex*)alpha, (const cuDoubleComplex* const*)A, lda, (const cuDoubleComplex* const*)x, incx, (const cuDoubleComplex*)beta, (cuDoubleComplex* const*)y, incy, batch_count);
+  return cudaSuccess;
+}
+
+
+template<typename T, int ROWBS, int COLBS>
+__global__ void ger_batched_kernel(const int m, // number of columns in row major A
```

Comment:
This is one important operation than MAGMA lacks, otherwise I'd argue we should use it.  Unfortunately it can also be a bit of a POS to build on some systems so...

Reply:
We can potentially ask the MAGMA folks to develop this via ECP as a hedge against incomplete vendor accelerated BLAS implementations.

Reply:
https://icl.cs.utk.edu/projectsfiles/magma/doxygen/group__magma__ger__batched.html ?